src/roktManager.ts

```diff
@@ -0,0 +1,86 @@
+// https://docs.rokt.com/developers/integration-guides/web/library/attributes
+export interface IRoktPartnerAttributes {
+    [key: string]: string | number | boolean | undefined | null;
+}
+
+// https://docs.rokt.com/developers/integration-guides/web/library/select-placements-options
+export interface IRoktSelectPlacementsOptions {
+    attributes: IRoktPartnerAttributes;
+    identifier?: string;
+}
+
+interface IRoktPlacement {}
+
+export interface IRoktSelection {
+    close: () => void;
+    getPlacements: () => Promise<IRoktPlacement[]>;
+}
+
+export interface IRoktLauncher {
+    selectPlacements: (options: IRoktSelectPlacementsOptions) => Promise<IRoktSelection>;
+}
+
+export interface IRoktMessage {
+    methodName: string;
+    payload: any;
+}
+
+export default class RoktManager {
+    public launcher: IRoktLauncher | null = null;
+    private messageQueue: IRoktMessage[] = [];
+
+    constructor() {
+        this.launcher = null;
+    }
+
+    public attachLauncher(launcher: IRoktLauncher): Promise<void> {
+        return new Promise<void>((resolve) => {
+            if (!launcher) {
+                this.queueMessage({
+                    methodName: 'attachLauncher',
+                    payload: launcher,
+                });
```

Comment:
this can be removed.  the expectation is there is no launcher and we are setting.